extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -314,26 +322,20 @@ function getAnchorElements(rootElement, anchorObj) {
 /**
  * @param {!Element} anchorElement
  * @param {!Position} position
- * @param {!./ancestor-blacklist.AncestorBlacklist} ancestorBlacklist
  * @return {boolean}
  */
-function isPositionValid(anchorElement, position, ancestorBlacklist) {
-  let ancestorBlacklisted = false;
-  if (position == Position.BEFORE || position == Position.AFTER) {
-    const parent = anchorElement.parentNode;
-    if (!parent) {
-      dev().warn(TAG, 'Parentless anchor with BEFORE/AFTER position.');
-      return false;
-    }
-    ancestorBlacklisted =
-        ancestorBlacklist.isOrDescendantOfBlacklistedElement(parent);
-  } else {
-    ancestorBlacklisted =
-        ancestorBlacklist.isOrDescendantOfBlacklistedElement(anchorElement);
-  }
-  if (ancestorBlacklisted) {
-    dev().warn(TAG, 'Placement inside blacklisted ancestor.');
+function isPositionValid(anchorElement, position) {
+  const elementToCheck = Position.BEFORE || position == Position.AFTER ?
+      anchorElement.parentElement : anchorElement;
+  if (!elementToCheck) {
+    dev().warn(TAG, 'Parentless anchor with BEFORE/AFTER position.');
     return false;
   }
-  return true;
+  return !BLACKLISTED_ANCESTOR_TAGS.some(tagName => {
+    if (closestByTag(/** @type {!Element} */ (elementToCheck), tagName)) {
```

Comment:
You have to reassign to a variable if you want that. Only the return value is explicitly type validated, not the input (which is the same as value as the return value). So something like:

```js
if (!element) {
  // yada yada
}
const defAnElement = dev().assertElement(element);
closestByTag(defAnElement, tagName)
```